gwpy/timeseries/timeseries.py

```diff
@@ -1290,6 +1290,41 @@ def rms(self, stride=1):
         return self.__class__(data, channel=self.channel, t0=self.t0,
                               name=name, sample_rate=(1/float(stride)))
 
+    def demod(self, f, stride=1):
```

Comment:
@alurban, why not call it `demodulate()`?

Reply:
@duncanmmacleod, this is a good suggestion and is taken care of in a new commit (https://github.com/gwpy/gwpy/pull/705/commits/bd0cbc6f3ae1715b6e5f8474746c5900a7335cc0)